universal/error-handling.md

```diff
@@ -0,0 +1,149 @@
+# Universal / Error handling
+
+An error can be categorized by many criteria, whether
+- it's **critical software** or it's **non-critical software**
+- it's **user facing** or **not user facing**
+- there **is a fallback** ir **there is no fallback**
+- it's on the **frontend** or **backend**
+- the code is **synchronous** or **asynchronous**
+- a certain language or library is used
+
+These are general rules aimed to help making a decision about how to handle certain errors. For errors texts refer to the [wording guideline](https://github.com/c-hive/guides/issues/2).
+
+## Critical software
+
+There's always a fallback strategy. Errors are always reported. Possible bugs are considered for graceful error handling.
+
+- Example: bank system
+
+## Non-critical software
+
+Bugs are not considered for graceful error handling. There isn't always a fallback strategy but errors should be reported.
+
+## User facing error
+
+Often happens on the frontend.
+
+#### By default, the user must be notified and a fallback strategy must be implemented.
+
+- Example: a form validation failed on post. The validation error is shown, the field is higlighted and the previous value may or may not be removed.
+
+- Example: worst case, some unexpected error happens. A generic error is shown: "Something unexpected happened, save your changes offline and [reload the page]()."
```

Comment:
Btw, what's the reason for using links in the case of the occurrences of `reload the page`? Redirects the viewer to another page.

Reply:
Hmmm that's very interesting, that's supposed to be an empty link but it redirects to one level up. I have no clue why, it might be related to github's markdown implementation.

Reply:
I think this is a bug. Wanted to report it, turns out GH doesn't have an open issue tracking system, which is.. ironic :D